addons/l10n_de/models/ir_attachment.py

```diff
@@ -14,7 +14,8 @@ class IrAttachment(models.Model):
     @api.ondelete(at_uninstall=True)
     def _except_audit_trail(self):
         audit_trail_attachments = self.filtered(lambda attachment:
-            attachment.res_model == 'account.move'
+            attachment.company_id.check_account_audit_trail                                        
+            and attachment.res_model == 'account.move'
```

Comment:
If you swap the order, you can limit your change to a single line keeping the file history a little more clear :+1:

```suggestion
            attachment.res_model == 'account.move'
            and attachment.company_id.check_account_audit_trail
```
I would even put it just before the `guess_mimetype` but that's your choice ultimately :wink: